### ISSUE TITLE
Image support

### ISSUE THREAD

**Issue by pedroteixeira** [evidenceprime/html-docx-js#2]
I just realized that docx that does not supported embed data url in <img src>. Do you have any ideas for this?

Was looking in a way to add imagens in the zip and reference them.. but seems tricky.
(I've done saving images in the zip, but not sure how to alter the altchunk)

thanks.

**Comment by anowak** [comment]
To be honest, I've never tried embedding images. I will look into this over the weekend.

**Comment by pedroteixeira** [comment]
We can save the images into the zip, using something like:
```
   var imgRE = /src="?([^"\s]+)"?\s*/g;   
   var canvas = document.createElement('canvas');
    var context = canvas.getContext('2d');

    while ( m = imgRE.exec( html ) ) {
      var url = m[1];
      if(url in images) {
        continue;
      }

      var image = new Image();
      image.src = url;
      context.drawImage(image, 0, 0);
      images[url] = canvas.toDataURL("image/png");
    }

    Object.keys(images).forEach(function(url) {
      html = html.replace(new RegExp('src="'+url+'"','g'), 'src="' + images[url] + '"');
    });
```

But after it seems a bit tricky.. because I would had to split the html into multiple Altchunks and create Word ML for each image. But it would be really cool to be able to solve it.

**Comment by SkyN** [comment]
Insert img - it is important for me too.

I test insert img to html (data uri):
1) < img src='data:image/gif;base64,R0lGODlhEAAQA[...skip...]IQA7' />
2) <style>div {background-image:url(data:image/gif;base64,R0lGODlhEAAQA[...skip...]IQA7)}</style>
IE 11 and Chrome 43 show img good, but word 2010 don't show it :(

**Comment by N1kto** [comment]
I've been looking into the problem recently. It appears to be that there is no way of processing images via `AltChunk`. 
`AltChunk` is **a temporary object** for a content (*html*) import to the word ml document. It is converted to *xml* by Word upon the first `save` action. `AltChunk` source is injected as content only, it doesn't tie any local resources (e.g. style sheets, images, etc).

The only workaround is to package the *html* source into *mhtml* file and serve it in *.doc* extension.

**Comment by N1kto** [comment]
OK. Kind of good news here. There is possibility to serve *mht* content via `altChunk`. 
The release implementing this will be available somewhere next week. Currently it is targets to support the embedded images only (*base64* injected via `DATA URI`), the support for 'regular' images (sourced from static) is also planned.

**Comment by anowak** [comment]
Fixed by 936f78b

**Comment by ashesnz** [comment]
Hi there,

Just tried running the demo with the fixed commit id (1d9fba...), however I am running into the same issue as SkyN (Word informs me this image cannot currently be displayed).

Version of Chrome I am using is 45 and Word 2010. (Image I am using is the cat.jpg as well)

Much appreciated with any help.
